atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/paging/RowGetter.java

```diff
@@ -59,6 +62,8 @@ public RowGetter(
     public List<KeySlice> getRows(KeyRange keyRange) throws Exception {
         ColumnParent colFam = new ColumnParent(CassandraKeyValueService.internalTableName(tableRef));
         InetSocketAddress host = clientPool.getRandomHostForKey(keyRange.getStart_key());
+        Set<InetSocketAddress> preferredBackupHosts = ImmutableSet.copyOf(
+                clientPool.getHostsForKey(keyRange.getStart_key()).orElse(ImmutableList.of()));
```

Comment:
Just curious here: you're getting the host that has the first key in the range, but the range might be quite large, is that right? Seems like the only sensible thing to do if so, I just wanted to check I understood everything correctly.